br/pkg/streamhelper/advancer.go

```diff
@@ -403,24 +470,24 @@ func (c *CheckpointAdvancer) subscribeTick(ctx context.Context) error {
 
 func (c *CheckpointAdvancer) importantTick(ctx context.Context) error {
 	c.checkpointsMu.Lock()
-	c.setCheckpoint(c.checkpoints.MinValue())
+	c.setCheckpoint(ctx, c.checkpoints.Min())
```

Comment:
... Also, when the async operation get blocked, there isn't a time out for importantTick, so perhaps we'd better move the lock resolver to another thread...